### PR TITLE
Testcase for gcc package

### PR DESCRIPTION
**Pull request by ramyabs**
This test-case make use of test-suite provided by the  gcc source package.

Signed-off-by: Ramya BS <ramya@linux.vnet.ibm.com>
Tested-by     : Ramya BS <ramya@linux.vnet.ibm.com>

### CODE REVIEW

File: toolchain/gcc.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                             'texinfo-tex',
+                             'texinfo',
+                             'elfutils-libelf-devel',
+                             'gmp-devel',
+                             'mpfr-devel',
+                             'libmpc-devel',
+                             'gcc-gnat',
+                             'libgnat'])
+        else:
+            self.log.warn("Unsupported OS!")
```
Comment (PraveenPenguin) [inline toolchain/gcc.py]: as after above comment it not required

File: toolchain/gcc.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        Runs the gcc `make check`
+        """
+        ret = process.run("make check", ignore_status=True, sudo=True)
+        logfile_name = os.path.join(self.logdir, "stdout")
+        logfile = open(logfile_name)
+        for line in logfile:
+            if line.startswith("\t\t==="):
+                self.log.info(line)
+            if line.startswith("#"):
+                self.log.info(line)
```
Comment (PraveenPenguin) [inline toolchain/gcc.py]: can you please close log file reference as well 

`logfile.close()`

File: toolchain/gcc.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        self.srcdir = os.path.join(self.srcdir, gcc_folder)
+        os.chdir(self.srcdir)
+        process.run('./configure', ignore_status=True, sudo=True)
+        build.make(self.srcdir)
+
+    def test(self):
+        """
+        Runs the gcc `make check`
+        """
+        ret = process.run("make check", ignore_status=True, sudo=True)
```
Comment (PraveenPenguin) [inline toolchain/gcc.py]: can you please use` build.run_make(self.srcdir,extra_args='check')`

File: toolchain/gcc.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        logfile_name = os.path.join(self.logdir, "stdout")
+        logfile = open(logfile_name)
+        for line in logfile:
+            if line.startswith("\t\t==="):
+                self.log.info(line)
+            if line.startswith("#"):
+                self.log.info(line)
+        if ret.exit_status:
+            self.fail("Few gcc tests failed,refer the log file")
```
Comment (PraveenPenguin) [inline toolchain/gcc.py]: can we parse log file as well and show fail point .Not any strong opinion but if we have this it will helpful

**Comment by ramyabs** [comment]
Raised new pull request for the same  https://github.com/avocado-framework-tests/avocado-misc-tests/pull/117 hence closing this one

**Pull request closed without merging**